tools/souffle_converter.py

```diff
@@ -1046,14 +1047,87 @@ def convert_terms(self, terms):
                 result.append(term)
         return result + tail
 
+    def flatten(self, tail, listbody, bool_dis):
+        for x in tail.children:
+            if (x.symbol.name == "Term"):
+                if (getOptField(x, "Literal") != None) or (getOptField(x, "!") != None):
+                    listbody.append(x)
+                child = getOptField(x, "Disjunction")
+                if child != None:
+                    if getOptField(child, "OR") != None:
+                        bool_dis = True
+                    listbody.append(child)
+            elif (x.symbol.name in ["OR", "(", ")", ",", "!"]):
+                listbody.append(x)
+            else:
+                (_, bool_dis) = self.flatten(x, listbody, bool_dis)
+        return (listbody, bool_dis)
+
+    def findCommas(self, comma):
+        commas = False
```

Comment:
In general I find it more readable to call such boolean variables hasCommas (or isXXX). Same for your bool_XXX naming above.